src/interrupt.ts

```diff
@@ -0,0 +1,119 @@
+}
+
+export function createInterruptHook(
+  toolConfigs: ToolInterruptConfig,
+  messagePrefix: string = "Tool execution requires approval",
+): (state: DeepAgentStateType) => Promise<DeepAgentStateType> {
+  /**
+   * Create a post model hook that handles interrupts using native LangGraph schemas.
+   *
+   * Args:
+   *   toolConfigs: Record mapping tool names to HumanInterruptConfig objects
+   *   messagePrefix: Optional message prefix for interrupt descriptions
+   */
+
+  return async function interruptHook(
+    state: DeepAgentStateType,
+  ): Promise<DeepAgentStateType> {
+    const messages = state.messages || [];
+    if (!messages.length) {
+      return state;
+    }
+
+    const lastMessage = messages[messages.length - 1] as AIMessage;
+    if (!lastMessage.tool_calls || !lastMessage.tool_calls.length) {
+      return state;
+    }
+
+    // Separate tool calls that need interrupts from those that don't
+    const interruptToolCalls: any[] = [];
+    const autoApprovedToolCalls: any[] = [];
+
+    for (const toolCall of lastMessage.tool_calls) {
+      const toolName = toolCall.name;
+      if (toolName in toolConfigs) {
+        interruptToolCalls.push(toolCall);
+      } else {
+        autoApprovedToolCalls.push(toolCall);
+      }
+    }
+
+    if (!interruptToolCalls.length) {
+      return state;
+    }
+
+    const approvedToolCalls = [...autoApprovedToolCalls];
+
+    const requests: HumanInterrupt[] = [];
+
+    for (const toolCall of interruptToolCalls) {
+      const toolName = toolCall.name;
+      const toolArgs = toolCall.args;
+      const description = `${messagePrefix}\n\nTool: ${toolName}\nArgs: ${JSON.stringify(toolArgs, null, 2)}`;
+      const toolConfig = toolConfigs[toolName];
+
+      const request: HumanInterrupt = {
+        action_request: {
+          action: toolName,
+          args: toolArgs,
+        },
+        config: toolConfig,
+        description: description,
+      };
+      requests.push(request);
+    }
+
+    const responses: HumanResponse[] = await interrupt(requests);
+
+    for (let i = 0; i < responses.length; i++) {
+      const response = responses[i];
+      const toolCall = interruptToolCalls[i];
+
+      if (response.type === "accept") {
+        approvedToolCalls.push(toolCall);
+      } else if (response.type === "edit" && response.args) {
+        const edited = response.args;
+        const newToolCall = {
+          name: toolCall.name,
+          args: edited.args,
+          id: toolCall.id,
+        };
+        approvedToolCalls.push(newToolCall);
+      } else if (response.type === "ignore") {
+        continue;
+      } else {
+        throw new Error(`Unknown response type: ${response.type}`);
+      }
+    }
+
+    const updatedLastMessage = {
+      ...lastMessage,
+      tool_calls: approvedToolCalls,
+    };
+
+    const updatedMessages = [...messages.slice(0, -1), updatedLastMessage];
+
+    return { ...state, messages: updatedMessages as typeof messages };
```

Comment:
same as above, shouldn't need to include state if it's acting as a node
```suggestion
    return { messages: updatedMessages as typeof messages };
```

Reply:
and if so, you should only need to return the new `updatedLastMessage`, not the entire messages list (since it has the same id as before the update, it should take precedence in the reducer)